### PR TITLE
core[minor]: Unified model params for LS

### PR DESCRIPTION
**Pull request by bracesproul**
openai[minor]
anthropic[minor]
community[minor]
cohere[minor]
mistral[minor]

### CODE REVIEW

File: langchain-core/src/language_models/chat_models.ts [hunk 1]
```diff
@@ -359,7 +388,7 @@ export abstract class BaseChatModel<
       handledOptions.tags,
       this.tags,
       handledOptions.metadata,
-      this.metadata,
+      inheritableMetadata,
```
Comment (nfcampos) [inline langchain-core/src/language_models/chat_models.ts:391]: this is the non inheritable metadata I think?

Comment (nfcampos) [reply langchain-core/src/language_models/chat_models.ts:391]: and in py I think we put it as inheritable

Comment (bracesproul) [reply langchain-core/src/language_models/chat_models.ts:390]: You are correct. Fixed in recent commit, thanks

File: libs/langchain-cohere/src/chat_models.ts [hunk 2]
```diff
@@ -145,6 +146,15 @@ export class ChatCohere<
     this.streaming = fields?.streaming ?? this.streaming;
   }
 
+  protected getLsParams(_: this["ParsedCallOptions"]): LangSmithParams {
+    return {
+      ls_provider: "cohere",
+      ls_model_name: this.model,
+      ls_model_type: "chat",
+      ls_temperature: this.temperature ?? undefined,
```
Comment (nfcampos) [inline libs/langchain-cohere/src/chat_models.ts:154]: this seems to be missing a bunch of params like max tokens, stop, etc see https://docs.cohere.com/reference/chat

File: libs/langchain-community/src/chat_models/ollama.ts [hunk 3]
```diff
@@ -176,6 +177,16 @@ export class ChatOllama
     this.format = fields.format;
   }
 
+  protected getLsParams(_: this["ParsedCallOptions"]): LangSmithParams {
+    return {
+      ls_provider: "ollama",
+      ls_model_name: this.model,
+      ls_model_type: "chat",
+      ls_temperature: this.temperature ?? undefined,
```
Comment (nfcampos) [inline libs/langchain-community/src/chat_models/ollama.ts:185]: no max tokens?

File: libs/langchain-mistralai/src/chat_models.ts [hunk 4]
```diff
@@ -417,6 +418,17 @@ export class ChatMistralAI<
...
   }
 
+  protected getLsParams(options: this["ParsedCallOptions"]): LangSmithParams {
+    const params = this.invocationParams(options);
+    return {
+      ls_provider: "mistral",
+      ls_model_name: this.model,
+      ls_model_type: "chat",
+      ls_temperature: params.temperature ?? undefined,
+      ls_max_tokens: params.maxTokens ?? undefined,
```
Comment (nfcampos) [inline libs/langchain-mistralai/src/chat_models.ts:428]: no stop seq?

Comment (bracesproul) [reply libs/langchain-mistralai/src/chat_models.ts:428]: correct. no support

**Pull request merged**